src/powerquery-language-services/analysis/documentAnalysis.ts

```diff
@@ -4,20 +4,19 @@
 import type { Position, Range } from "vscode-languageserver-types";
 import type { TextDocument } from "vscode-languageserver-textdocument";
 
-import { WorkspaceCache, WorkspaceCacheUtils } from "../workspaceCache";
 import { AnalysisBase } from "./analysisBase";
 import { AnalysisSettings } from "./analysisSettings";
+import { WorkspaceCacheUtils } from "../workspaceCache";
 
 export class DocumentAnalysis extends AnalysisBase {
     constructor(private readonly textDocument: TextDocument, analysisSettings: AnalysisSettings, position: Position) {
         super(
             analysisSettings,
-            WorkspaceCacheUtils.getOrCreateInspection(
+            WorkspaceCacheUtils.getOrCreateInspectionPromise(
```

Comment:
I'm still thinking we'll eventually want to eliminate the workspace cache entirely, but this works in the interim.

Reply:
See comment above on my reasoning for the WorkspaceCache.